File: fieldsync-inventory-backend/src/main/java/com/example/fieldsync_inventory_api/dto/inventory/seed_batch/SeedBatchResponseDTO.java
package com.example.fieldsync_inventory_api.dto.inventory.seed_batch;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.example.fieldsync_inventory_api.dto.inventory.rice_generation.RiceGenerationCompactDTO;
import com.example.fieldsync_inventory_api.dto.inventory.rice_variety.RiceVarietyCompactDTO;
import com.example.fieldsync_inventory_api.dto.inventory.season.SeasonCompactDTO;

import java.time.Instant;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SeedBatchResponseDTO {
    private Long id;
    private RiceVarietyCompactDTO variety;
    private RiceGenerationCompactDTO generation;
    private SeasonCompactDTO season;
    private Integer year;
    private Boolean grading;
    private Boolean germination;
    private String description;
    private Instant createdAt;
    private Instant updatedAt;
    private Instant deletedAt;
}

File: fieldsync-inventory-app/app/src/main/java/com/example/fieldsync_inventory_app/data/remote/dto/seed_batch/SeedBatchResponseDto.kt
package com.example.fieldsync_inventory_app.data.remote.dto.seed_batch

data class SeedBatchResponseDto(
    val id: Long,
    val variety: IdNamePair,
    val generation: IdNamePair,
    val season: IdNamePair,
    val year: Int,
    val grading: Boolean,
    val germination: Boolean,
    val description: String?,
    val createdAt: String,
    val updatedAt: String,
    val deletedAt: String?
)